### PR TITLE
Semicolon Linting Fix

### PR DESCRIPTION
**Pull request by lukeburpee**
### Problem
The generator does not update the tslint.json file when the semicolon option is changed in the generator spec, resulting in linting errors. This request updates the writing generator to address this issue. The request also updates the writing generator index to allow semicolon changes to the generated channels and log hook files. Finally, this request addresses a semicolon error in generated js applications where a semicolon should only be generated for typescript applications when importing the typescript app interface in the seed-data file.

### Open Issues
No current issues

### Related Pull Requests
This request relates to pull request feathers-plus#83, providing the semicolon linting fix portion of the original request. The development environment implementation from request feathers-plus#83, along with updates to generator tests, are provided in separate pull requests

### Note
tests require all three requests related to #83 to pass

### CODE REVIEW

**Comment by eddyystop** [comment]
If you see the same comment multiple times, thank pjax. Just refresh the page. I used to get phenominal performance with pjax, but you keep running into issues like this. More importantly it often breaks on page back and forward.

**Comment by eddyystop** [comment]
Published as cli+ 0.7.71 which run generator v0.6.69. Thanks!
